### DISCUSSION TITLE
Create zero-config profiles/modes

### DISCUSSION THREAD

**Discussion by ORESoftware** [verdaccio/verdaccio#972] [Ideas]
This is especially useful when using verdaccio with Docker, I would imagine, because containers cannot read the host fs, and users are not going to want to open up the container to modify config files.

In general, the yaml config for verdaccio is confusing and unknown to newcomers.

What would be ideal is if there were profiles/preconfig for the most common use cases. I am looking for a profile that does not require auth:

```bash
verdaccio --simple-proxy --no-auth
```

with the above, the command line options override the config (naturally). All I want is no auth required for the local registry and I want uploads to only go to local registry and downloads to start at the local registry and only get forwarded to global registry if the package cannot be found in local registry.

As a side note, I find yaml to be a nightmare and if json config could be an option that would nice.

**Comment by dlouzan** [comment]
I'm not sure I agree with this suggestion, I see the point of simplifying the configuration or making it more understandable. But adding startup profiles that ignore the configuration file settings? Now, *that* would be confusing.

What is the advantage of that compared to, let's say, a good library of sample configurations for such use cases in the documentation?

**Comment by ORESoftware** [comment]
@dlouzan for most command line tools, the command line options override hardcoded configuration, it's completely standard and expected.

**Comment by dlouzan** [comment]
@ORESoftware  Wow, what a condescending answer.

The proposal is not to be able to overwrite the same settings in the configuration file as command-line options, or maybe a generic mode for being able to load the configuration from different sources, à la Spring Boot and its PropertyLocator. As I understand it, it is more or less that functionality, plus extra options that load specific values for subsets of those configurations.

So I ask again, what is the advantage of profiles over having a good set of examples in the documentation for the different scenarios? verdaccio is not an end-user application, it's a support tool for development teams.

**Comment by ORESoftware** [comment]
@dlouzan surprise! i want to use it as part of an end user application, for usage with libraries. :)

And my previous answer was not condescending, it was to the point - I would like a zero-config option that doesnt require env variables or config files if possible.
